frontend/src/metabase/components/Timeline/Timeline.tsx

```diff
@@ -0,0 +1,107 @@
+import React from "react";
+import _ from "underscore";
+import { t } from "ttag";
+import { getRelativeTime } from "metabase/lib/time";
+
+import type { Revision } from "metabase-types/api";
+import Button from "metabase/core/components/Button";
+import Tooltip from "metabase/core/components/Tooltip";
+
+import {
+  TimelineContainer,
+  TimelineEvent,
+  Border,
+  StyledIcon,
+  EventBody,
+  EventHeader,
+  Timestamp,
+} from "./Timeline.styled";
+
+const ICON_SIZE = 16;
+const HALF_ICON_SIZE = ICON_SIZE / 2;
+
+type Icon = string | { name: string; color: string } | Record<string, never>;
+export type TimelineEvent = {
```

Comment:
`metabase/components` is a deprecated folder. If we're creating a domain-specific component that should be reused across features, it should go to `metabase/common/components`